Makefile

```diff
@@ -298,84 +311,84 @@ $(O_FILES): | asset_files
 
 .PHONY: o_files asset_files
 
-build/$(SPEC): $(SPEC)
-	$(CPP) $(CPPFLAGS) $< > $@
+$(BUILD_DIR)/$(SPEC): $(SPEC)
+	$(CPP) $(CPPFLAGS) $< | sed -e 's|$$(BUILD_DIR)|$(BUILD_DIR)|g' > $@
```

Comment:
I think it would be cleaner to use cpp (the C preprocessor) instead, for example

```suggestion
	$(CPP) $(CPPFLAGS) '-DBUILD_DIR=$(BUILD_DIR)' $< > $@
```
and use `BUILD_DIR` instead of `$(BUILD_DIR)` in the spec

Wait for more input to make this change though. There are alternatives like having the spec use `build/VERSION/` which may be discussed too

Reply:
CPP does not replace macros in string literals 🙁

Reply:
Oh right the spec format is `include "path/to/tu.o"`... I didn't realize that cpp would see it as a string
Bleh
(ftr I'm a long time hater of the spec format <img src="https://cdn.discordapp.com/emojis/943102210366119956.webp?size=56&quality=lossless" alt="skawoUHHUH" height="24"> )